table_schema_cache.go

```diff
@@ -189,15 +190,12 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 			tableLog := dbLog.WithField("table", tableName)
 			tableLog.Debug("caching table schema")
 
-			// Sanity check
-			if len(tableSchema.PKColumns) != 1 {
-				err = fmt.Errorf("table %s has %d primary key columns and this is not supported", tableName, len(tableSchema.PKColumns))
-				logger.WithError(err).Error("invalid table")
+			if err := tableSchema.setPaginationKeyColumn(dbname, tableName, paginateByColumnConfig); err != nil {
 				return tableSchemaCache, err
 			}
 
 			if tableSchema.GetPKColumn(0).Type != schema.TYPE_NUMBER {
```

Comment:
I know that you override the `GetPKColumn` and I spoke with you about it, but this is pretty confusing now that I see it. I think we should change this if it is not too difficult.

Reply:
Yes I agree. I will change it as a separate pull request. I want to keep this pull request relevant to the functionality required.